src/messages/MessageBuilder.cpp

```diff
@@ -385,8 +385,26 @@ EmotePtr makeAutoModBadge()
         Url{"https://dashboard.twitch.tv/settings/moderation/automod"}});
 }
 
-EmotePtr makeSharedChatBadge(const QString &sourceName)
+EmotePtr makeSharedChatBadge(const QString &sourceName, const QString &url)
 {
+    if (!url.isEmpty())
+    {
+        QString modifiedUrl = url;
+
+        modifiedUrl.replace("300x300", "70x70");
+
+        return std::make_shared<Emote>(Emote{
+            .name = EmoteName{},
+            .images = ImageSet{Image::fromUrl(
+                Url{modifiedUrl},
+                0.257142857143)},  // get as close to 18x18 as possible
+            .tooltip =
+                Tooltip{"Shared Message" +
+                        (sourceName.isEmpty() ? "" : " from " + sourceName)},
+            .homePage = Url{"https://link.twitch.tv/SharedChatViewer"},
```

Comment:
We could link to the channel here (but we need to pass the login name in that case).

Reply:
Looks like we don't support the homePage thing on badges atm - right-clicking a badge doesn't give me any link options

Reply:
I updated the link in [`88b65e6` (#5760)](https://github.com/Chatterino/chatterino2/pull/5760/commits/88b65e61568d88ca7d110213df332de7cf9afbcd) anyway - won't try to fix the right-click action for badge elements in this PR though